File: java/Calculator.java
import java.util.Scanner;


public class Calculator {

    public Calculator(){};

    public static int add(int a, int b){
        return a + b;
    }

    public static int subtract(int a, int b){
        return a - b;
    }

    public static int multiply(int a, int b){
        return a * b;
    }

    public static int divide(int a, int b){
        return a / b;
    }

    public static int modulus(int a, int b){
        return a % b;
    }






public static void main(String[] args) {
    Scanner scan = new Scanner(System.in);
    int a,b;
    String option;
    Calculator calculator = new Calculator();

    System.out.print("Enter first integer: ");
    a = scan.nextInt();
    System.out.print("Enter second integer: ");
    b = scan.nextInt();
    System.out.println("Pick an option below:");
    System.out.println("******\t Pick Arthmetic operation\t*******");
    System.out.print("Add: Enter 'add' \nSubtract: Enter 'sub' \nMultiply: Enter 'mult' \nDivide: Enter 'div' \nModulus/Remainder: Enter 'mod' \n");
    System.out.println("******************************************");
    System.err.print("Option: ");
    option = scan.next();

    switch (option) {
        case "mod":
            System.out.print("Modulus of " + a + " and " + b + " is ");
            System.out.print(calculator.modulus(a, b));
            System.out.println();
            break;
        case "add":
            System.out.print("Addition of " + a + " and " + b + " is ");
            System.out.print(calculator.add(a, b));
            System.out.println();
            break;
        case "sub":
            System.out.print("Subtraction of " + a + " and " + b + " is ");
            System.out.print(calculator.subtract(a, b));
            System.out.println();
            break;
        case "mult":
            System.out.print("Subtraction of " + a + " and " + b + " is ");
            System.out.print(calculator.multiply(a, b));
            System.out.println();
            break;
        case "div":
            System.out.print("Subtraction of " + a + " and " + b + " is ");
            System.out.print(calculator.divide(a, b));
            System.out.println();
            break;
        default:
            throw new AssertionError();
    }
}


}

File: rust/Calculator.rs
use std::io;

struct Calculator;

impl Calculator {
    fn add(a: i32, b: i32) -> i32 {
        a + b
    }
    fn sub(a: i32, b: i32) -> i32 {
        a - b
    }
    fn multiply(a: i32, b: i32) -> i32 {
        a * b
    }
    fn divide(a: i32, b: i32) -> i32 {
        a / b
    }
    fn modulus(a: i32, b: i32) -> i32 {
        a % b
    }
}

fn main() {
    let mut x = String::new();
    let mut y = String::new();
    let mut opt = String::new();

    println!("Enter the first variable: ");
    io::stdin()
        .read_line(&mut x)
        .expect("Failed to read line");
    let x: i32 = x.trim().parse().expect("Please type a number!");

    println!("Enter the second variable: ");
    io::stdin()
        .read_line(&mut y)
        .expect("Failed to read line");
    let y: i32 = y.trim().parse().expect("Please type a number!");

    println!("Enter the options from below:\nAdd: Enter add\nSub: Enter sub\nMultiply: Enter mult\nDivide: Enter div\nModulus: Enter mod\nOption:");
    io::stdin()
        .read_line(&mut opt)
        .expect("Failed to read line");
    let opt = opt.trim();

    match opt {
        "add" => println!("The result of adding {} and {} is {}", x, y, Calculator::add(x, y)),
        "sub" => println!("The result of subtracting {} and {} is {}", x, y, Calculator::sub(x, y)),
        "mult" => println!("The result of multiplying {} and {} is {}", x, y, Calculator::multiply(x, y)),
        "div" => println!("The result of dividing {} and {} is {}", x, y, Calculator::divide(x, y)),
        "mod" => println!("The result of modulus {} and {} is {}", x, y, Calculator::modulus(x, y)),
        _ => println!("Invalid option!"),
    }
}